test/test_euler.py

```diff
@@ -0,0 +1,874 @@
+            logging.info(f"Max error: {maxxerr}")
+            message = (
+                f"Error for (dim,order) = ({dim},{order}):\n"
+                f"{eoc_rec}"
+            )
+            logging.info(message)
+            assert (
+                eoc_rec.order_estimate() >= order - 0.5
+                or eoc_rec.max_error() < tolerance
+            )
+
+
+def test_isentropic_vortex():
+    """Advance the 2D isentropic vortex case in
+    time with non-zero velocities using an RK4
+    timestepping scheme. Check the advanced field
+    values against the exact/analytic expressions.
+
+    This tests all parts of the Euler module working
+    together, with results converging at the expected
+    rates vs. the order.
+    """
+    cl_ctx = cl.create_some_context()
+    queue = cl.CommandQueue(cl_ctx)
+
+    dim = 2
+
+    for order in [1, 2, 3]:
+        from pytools.convergence import EOCRecorder
+
+        eoc_rec = EOCRecorder()
+
+        for nel_1d in [16, 32, 64]:
+            from meshmode.mesh.generation import (
+                generate_regular_rect_mesh,
+            )
+
+            mesh = generate_regular_rect_mesh(
+                a=(-5.0,) * dim, b=(5.0,) * dim, n=(nel_1d,) * dim
+            )
+
+            t = 0
+            t_final = 0.01
+            istep = 0
+
+            discr = EagerDGDiscretization(cl_ctx, mesh, order=order)
+            nodes = discr.nodes().with_queue(queue)
+
+            vel = np.zeros(shape=(dim,))
+            orig = np.zeros(shape=(dim,))
+            vel[0] = 1.0
+            vel[1] = 1.0
+            initializer = Vortex2D(center=orig, velocity=vel)
+            #    initializer = Lump(center=orig,velocity=vel)
+            boundaries = {BTAG_ALL: PrescribedBoundary(initializer)}
+            eos = IdealSingleGas()
+
+            fields = initializer(0, nodes)
+
+            cfl = 1.0
+            sdt = get_inviscid_timestep(discr, fields, c=cfl, eos=eos)
+            constantcfl = False
+            dt = 0.001
+            nstep_status = 10
+
+            initname = initializer.__class__.__name__
+            eosname = eos.__class__.__name__
+            message = (
+                f"Num {dim}d elements: {mesh.nelements}\n"
+                f"Timestep:        {dt}\n"
+                f"Final time:      {t_final}\n"
+                f"Status freq:     {nstep_status}\n"
+                f"Initialization:  {initname}\n"
+                f"EOS:             {eosname}"
+            )
+            logging.info(message)
+
+            def write_soln():
+                expected_result = initializer(t, nodes)
+                result_resid = fields - expected_result
+                maxerr = [
+                    np.max(np.abs(result_resid[i].get()))
+                    for i in range(dim + 2)
+                ]
+                dv = eos(fields)
+                mindv = [np.min(dvfld.get()) for dvfld in dv]
+                maxdv = [np.max(dvfld.get()) for dvfld in dv]
```

Comment:
Use a norm and CL reductions. (Here and elsewhere.)

Reply:
Example needed.  I wasn't able to find any examples of this in Grudge, or PyOpenCL.

Reply:
(see above)